src/primitive_file_win.cc

```diff
@@ -154,32 +153,113 @@ PRIMITIVE(open) {
     // with open (eg a pipe, a socket, a directory).  We forbid this because
     // these file descriptors can block, and this API does not support
     // blocking.
-    INVALID_ARGUMENT;
+    if (strcmpi(R"(\\.\NUL)", pathname) != 0)
```

Comment:
I prefer adding { } here unless the call to INVALID_ARGUMENT is on the same line.